pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -547,55 +536,29 @@ pub mod pallet {
 			domain: Domain,
 			message: T::Message,
 		) -> (DispatchResult, Weight) {
-			let mut weight = T::DbWeight::get().reads(1);
+			let read_weight = T::DbWeight::get().reads(1);
 
-			let router = match DomainRouters::<T>::get(domain) {
-				Some(r) => r,
-				None => return (Err(Error::<T>::RouterNotFound.into()), weight),
+			let Some(router) = DomainRouters::<T>::get(domain) else {
+				return (Err(Error::<T>::RouterNotFound.into()), read_weight);
 			};
 
-			match router.send(sender.clone(), message.serialize()) {
-				Ok(dispatch_info) => Self::update_total_post_dispatch_info_weight(
-					&mut weight,
-					dispatch_info.actual_weight,
-				),
-				Err(e) => {
-					Self::update_total_post_dispatch_info_weight(
-						&mut weight,
-						e.post_info.actual_weight,
-					);
-
-					return (Err(e.error), weight);
-				}
-			}
+			let serialized = message.serialize();
+			let serialized_len = serialized.len() as u64;
 
-			(Ok(()), weight)
-		}
-
-		/// Updates the provided `PostDispatchInfo` with the weight required to
-		/// process an outbound message.
-		pub(crate) fn update_total_post_dispatch_info_weight(
-			weight: &mut Weight,
-			router_call_weight: Option<Weight>,
-		) {
-			let router_call_weight =
-				Self::get_outbound_message_processing_weight(router_call_weight);
-
-			*weight = weight.saturating_add(router_call_weight);
-		}
+			// TODO: do we really need to return the weights in send() if later we use the
+			// defensive ones?
+			let (result, router_weight) = match router.send(sender, serialized) {
+				Ok(dispatch_info) => (Ok(()), dispatch_info.actual_weight),
+				Err(e) => (Err(e.error), e.post_info.actual_weight),
+			};
```

Comment:
@cdamian, what do you think about this? Could we just convert `router.send()` to return just a simple `DispatchResult`?

cc @wischli

Reply:
Below, we're adding the defensive weights

Reply:
I would only return defensive weights when there's no way of finding the weight for an operation. In this case, the router does have some weight and I think we should use that, maybe alongside the defensive ones if we really wanna be extra-defensive?

Reply:
Mmmm... I see. The defensive one seems to already shadow anything returned by the router. So, I would either only use the router one or only use the defensive one.

But not super strong opinion

Reply:
I would also use either the "real" or the defensive weight. If the weights of `send` can be assumed to be representative, then let's go with that. Else, ignore it and fallback to the defensive one for now.